### PR TITLE
Generic toolbar buttons

### PR DESCRIPTION
**Pull request by fschmenger**
This adds a generic toggle button as a Wegue component, which replaces redundant implementations of each toggle button for the individual modules. As a consequence, the modules icon resource now has to be declared in app-conf.json.

### CODE REVIEW

**Comment by fschmenger** [comment]
This adds a generic toggle button as a Wegue component, which replaces redundant implementations of each toggle button for the individual modules. As a consequence, the modules icon resource now has to be declared in app-conf.json.

**Review by chrismayer — APPROVED** [review]
LGTM, thnaks for this @fschmenger. I added 2 minor (non-blocking) comments you might want to address before merging

**Comment by fschmenger** [comment]
Regarding this change:





-    <v-menu v-if="menuButtons.length" offset-y>

+    <v-menu v-if="menuButtons.length" offset-y nudge-bottom="15">


One implication of wrapping buttons by a toggle-group is, that the orientation of the hamburger menu is now vertical. So I inserted a little offset such that it is not showing on top of the toolbar. IMO it doesn’t look very nice without it, but I can certainly remove it.

geomer GmbH
Felix Schmenger
Entwicklung
Im Breitspiel 11 B
69126 Heidelberg
Germany

Tel.:+49 (6221) 89458-49
Fax: +49 (6221) 89458-79
E-Mail: ***@***.******@***.***>
URL: www.geomer.de<https://www.geomer.de/>

[partner]
[abobutton]<https://www.geomer.de/anmeldung-newsletter>

Firmensitz: Im Breitspiel 11 b, 69126 Heidelberg, Deutschland
Registergericht: Amtsgericht Mannheim, HRB 336266
Geschäftsführer: Martin Schroeder

File: app-starter/components/AppHeader.vue [hunk 1]
```diff
@@ -24,26 +24,29 @@
...
+          :dark="tbButton.dark" :moduleName="tbButton.moduleName"
       />
     </template>
 
     <!-- slot to inject components after the auto-generated buttons (by config) -->
     <slot name="wgu-tb-after-auto-buttons"></slot>
 
-    <v-menu v-if="menuButtons.length" offset-y>
+    <v-menu v-if="menuButtons.length" offset-y nudge-bottom="15">
       <!--v-slot="activator" to v-slot:activator="{on}" -->
```
[the remark this answers is not in the document]
Comment (fschmenger) [reply app-starter/components/AppHeader.vue:35]: Regarding this change:
-    <v-menu v-if="menuButtons.length" offset-y>
+    <v-menu v-if="menuButtons.length" offset-y nudge-bottom="15">

One implication of wrapping buttons by a toggle-group is, that the orientation of the hamburger menu is now vertical. So I inserted a little offset such that it is not showing on top of the toolbar. IMO it doesn’t look very nice without it, but I can certainly remove it.

[the remark this answers is not in the document]
Comment (chrismayer) [reply app-starter/components/AppHeader.vue:35]: My question was with regard to the commented line. Not the line above.

**Pull request merged**